pkg/signature/kms/kms.go

```diff
@@ -72,7 +72,14 @@ func SupportedProviders() []string {
 type SignerVerifier interface {
 	signature.SignerVerifier
 	CreateKey(ctx context.Context, algorithm string) (crypto.PublicKey, error)
-	CryptoSigner(ctx context.Context, errFunc func(error)) (crypto.Signer, crypto.SignerOpts, error)
+	CryptoSigner(ctx context.Context, errFunc func(error)) (CryptoSignerWrapper, crypto.SignerOpts, error)
```

Comment:
What's the motivation for this change over just exposing the additional `HashFunc` function? This would be a breaking change if anyone is expecting a crypto.Signer, correct?